src/ansys/fluent/core/solver/flobject.py

```diff
@@ -375,18 +375,25 @@ def find_object(self, relative_path):
                 obj = getattr(obj, comp)
         return obj
 
-    def before_execute(self, value):
+    def before_execute(self, command_name, value, kwargs):
         """Executes before command execution."""
         if hasattr(self, "_do_before_execute"):
-            self._do_before_execute(value)
-            return True
+            base_file_name = self._do_before_execute(
+                command_name=command_name, value=value, kwargs=kwargs
+            )
+            return base_file_name
```

Comment:
Just a question: is it only commands associated with file operations that can "do things" before command execution? The name `_do_before_execute` seems very generic to me which implies otherwise, but it returns `base_file_name` which is very file-operation specific.

Reply:
is it only commands associated with file operations that can "do things" before command execution? - Yes